packages/dd-trace/src/exporters/common/request.js

```diff
@@ -56,7 +56,11 @@ function request (data, options, callback) {
   if (options.url) {
     const url = typeof options.url === 'object' ? urlToOptions(options.url) : fromUrlString(options.url)
     if (url.protocol === 'unix:') {
-      options.socketPath = url.pathname
+      if (typeof options.url === 'string') {
+        options.socketPath = options.url.replace(/^unix:/, '')
```

Comment:
This should happen in `fromUrlString` since the check for a string has already been done at that point and it's responsible for the translation. There should also be a test added to ensure that this is actually working and regressions don't happen in the future.